src/kafka_clj_utils/producers.clj

```diff
@@ -40,42 +40,49 @@
          value-ser       (avro-serializer/->avro-serializer serde-config)]
      (KafkaProducer. ^Map producer-config key-ser value-ser))))
 
-(defn- ->failure-tracking-callback [failure-state]
+(defn- ->callback [failure-state ack-callback-fn]
   (reify org.apache.kafka.clients.producer.Callback
     (onCompletion [_this _metadata ex]
-      (when ex
-        (reset! failure-state ex)))))
+      (if ex
+        (reset! failure-state ex)
+        (ack-callback-fn)))))
```

Comment:
Is it worth passing the `metadata` to `ack-callback-fn`?
Looks like it has handy things in it which might not be useful right now but saves us another upgrade
https://kafka.apache.org/0100/javadoc/org/apache/kafka/clients/producer/RecordMetadata.html